purldb-toolkit/src/purldb_toolkit/purlcli.py

```diff
@@ -27,47 +29,129 @@ def purlcli():
+    help="Write versions output as JSON to FILE.",
 )
 @click.option(
     "--file",
     type=click.File(mode="r", encoding="utf-8"),
     required=False,
     help="Read a list of PURLs from a FILE, one per line.",
 )
-def validate(purls, output, file):
+def get_meta(purls, output, file):
     """
-    Check the syntax of one or more PURLs.
+    Given one or more PURLs, for each PURL, return a mapping of metadata
+    fetched from the fetchcode package.py info() function.
     """
-    if purls and file:
-        raise click.UsageError("Use either purls or file but not both.")
-
-    if not (purls or file):
-        raise click.UsageError("Use either purls or file.")
+    check_for_duplicate_input_sources(purls, file)
 
     if file:
         purls = file.read().splitlines(False)
 
-    validated_purls = validate_purls(purls)
-    json.dump(validated_purls, output, indent=4)
+    context = click.get_current_context()
+    manual_command_name = context.command.name
 
+    meta_info = get_meta_details(purls, output, file, manual_command_name)
+    json.dump(meta_info, output, indent=4)
+
+
+def get_meta_details(purls, output, file, manual_command_name):
+    """
+    Return a dictionary containing metadata for each PURL in the `purls` input
+    list.  `check_meta_purl()` will print an error message to the console
+    (also displayed in the JSON output) when necessary.
+    """
+    context_purls = [p for p in purls]
+    context_file = file
+    context_file_value = None
+    if context_file:
+        context_file_value = context_file.name
+
+    meta_details = {}
+    meta_details["headers"] = []
+    meta_details["packages"] = []
+
+    headers_content = {}
+    options = {}
+
+    errors = []
+    warnings = []
+
+    # TODO: How do we want to implement the try/except, the results of which will be reported in the JSON output?
+
+    headers_content["tool_name"] = "purlcli"
+    headers_content["tool_version"] = version("purldb_toolkit")
+
+    options["command"] = manual_command_name
+    options["--purl"] = context_purls
+    options["--file"] = context_file_value
+
+    if isinstance(output, str):
+        options["--output"] = output
+    else:
+        options["--output"] = output.name
+
+    headers_content["options"] = options
+    headers_content["purls"] = purls
+
+    meta_details["headers"].append(headers_content)
 
-def validate_purls(purls):
-    api_query = "https://public.purldb.io/api/validate/"
-    validated_purls = []
     for purl in purls:
+        meta_detail = {}
+        meta_detail["purl"] = purl
+        meta_detail["metadata"] = []
+
         purl = purl.strip()
         if not purl:
             continue
-        request_body = {"purl": purl, "check_existence": True}
-        response = requests.get(api_query, params=request_body)
-        results = response.json()
-        validated_purls.append(results)
 
-    return validated_purls
+        if check_meta_purl(purl):
```

Comment:
Assign `check_meta_purl(purl)` to a variable and replace all calls to `check_meta_purl(purl)` with it.

Reply:
Done.